### DISCUSSION TITLE
Steps to install React Router version 5

### DISCUSSION THREAD

**Discussion by akash-coded** [akash-coded/mern#175] [Guidelines]
Here are the updated steps, including the justification and instructions to install React Router version 5:

Step 1: Install React Router version 5
- Open your terminal and navigate to the project directory if you're not already there.
- Install React Router version 5 by running the command: `npm install react-router-dom@5.2.0`.

Justification:
React Router version 5 introduced some changes and improvements over the previous versions. It provides a more stable and consistent API, with better support for nested routes, route matching, and navigation. By using React Router version 5, you can ensure compatibility and take advantage of the latest features and improvements.

Step 2: Update the import statements in `App.js`
- Open the `src/App.js` file.
- Replace the import statement for React Router with the following code:
```jsx
import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom';
```

Step 3: Update the `<Router>` component in `App.js`
- Replace the `<Router>` component with the following code in the `App` function:
```jsx
<Router basename={process.env.PUBLIC_URL}>
```

Step 4: Add a 404 Not Found page
- Inside the `pages` directory, create a new file called `NotFound.js`.
- Open `NotFound.js` and add the following code:
```jsx
import React from 'react';

function NotFound() {
  return <h3>404 Page Not Found</h3>;
}

export default NotFound;
```

Step 5: Update the `<Switch>` component in `App.js`
- Add the following code inside the `<Switch>` component in `App.js`:
```jsx
<Route exact path="/">
  <Home products={products} addToCart={addToCart} />
</Route>
<Route path="/product/:id">
  <ProductDetail products={products} addToCart={addToCart} />
</Route>
<Route path="/cart">
  <Cart
    cart={cart}
    incrementQuantity={incrementQuantity}
    decrementQuantity={decrementQuantity}
    removeItem={removeItem}
  />
</Route>
<Route component={NotFound} />
```

Step 6: Update the navigation link in `App.js`
- Replace the navigation link in the navigation bar inside the `App` function with the following code:
```jsx
<Link to="/cart" className="nav-link">
  Cart ({cart.length})
</Link>
```

Step 7: Run the app
- Make sure the development server is running (`npm start`).
- Open your browser and go to `http://localhost:3000`.
- You should see the "Amazezon" heading, a navigation bar, and a list of products on the home page.
- Clicking on "Add to Cart" will add the product to the cart.
- The cart page will display the added items with quantity and options to increment, decrement, or remove them.
- The "Checkout" button is not functional in this exercise.
- Clicking on "View Details" will navigate to the product detail page.
- The product detail page will display the name, description, and an option to add the product to the cart.
- The "Back to Home" link will navigate back to the home page.
- Navigating to an undefined route will display the "404 Page Not Found" page.

Justification:
By installing React Router version 5, you ensure compatibility and stability with the latest version of React and other dependencies. This version provides enhanced features and a more refined API, allowing for better control and management of routing within your application.

Congratulations! You have now successfully updated the app to use React Router version 5. Users can navigate between different pages using the updated routing setup. If you encounter any issues or have further questions, feel free to ask. Enjoy exploring and shopping with your magical Amazezon app!

**Comment by kkpai** [comment]
![Amazezon](https://github.com/akash-coded/mern/assets/15033880/83dd201b-c7fb-40ad-ad9c-93e52fd05e60)

**Comment by Sagar1006-dev** [comment]
![2023-07-09 (1)](https://github.com/akash-coded/mern/assets/129708548/03190f60-ff26-4488-a7a3-cd21ead74fed)
